guide/additional-info/changes-in-v14.md

```diff
@@ -0,0 +1,176 @@
+
+```js
+const { EnumResolver } = require('discord.js');
+const enumValue = EnumResolvers.resolveButtonStyle('PRIMARY');
+```
+
+Areas like JSON slash commands and JSON message compononents will likely need to be modified to accomodate these changes:
+
+**Application commands**:
+
+```diff
++ const { ApplicationCommandType, ApplicationCommandOptionType } = require('discord.js');
+
+const command = {
+  name: 'ping',
+- type: 'CHAT_INPUT',
++ type: ApplicationCommandType.ChatInput,
+  options: [
+    name: 'option',
+    description: 'A sample option'
+-   type: 'STRING',
++   type: ApplicationCommandOptionType.String
+  ],
+};
+```
+
+**Buttons**:
+
+```diff
++ const { ButtonStyle } = require('discord.js');
+
+const button = {
+  label: 'test'
+- style: 'PRIMARY',
++ style: ButtonStyle.Primary,
+  customId: '1234'
+}
+```
+
+### Event Removals
+
+The `message` and `interaction` events are now removed. Use `messageCreate` and `interactionCreate` instead.
+
+`applicationCommandCreate`, `applicationCommandDelete` and `applicationCommandUpdate` have all been removed. Refer to [this pull request](https://github.com/discordjs/discord.js/pull/6492) for context.
+
+### REST Events
+
+The following discord.js events:
+- `invalidRequestWarning`
+- `request`
+- `response`
+- `rateLimited`
+- `newListener`
+- `removeListener`
+
+Have been removed from the `Client` in discord.js. Instead you should access these events from `Client#rest`.
+
+### Utility Changes
+
+`Util#removeMentions` has been removed, to control mentions you should use `allowedMentions` on `MessageOptions` instead.
+
+### CDN Changes
+
+Methods that return CDN URLs will now return a dynamic image URL (if available). This behavior can be overriden by setting `forceStatic` to `false` in the `MakeURLOptions` parameters.
+
+### Guild Object Changes
+
+`Guild#setRolePositions` and `Guild#setChannelPositions` have been removed. Use `RoleManager#setPositions` and `GuildChannelManager#setPositions` instead respectively.
+
+### Thread Member Manager
+
+`ThreadMemberManager#fetch` now only takes a single object of type `ThreadMemberFetchOptions`.
+
+### Role Manager
+
+`Role.comparePositions` has been removed. Use `RoleManager#comparePositions` instead.
+
+### `.deleted` Field(s) have been removed
+
+You can no longer use `#deleted` to check if a structure was deleted or not. 
+
+Check out [the issue ticket](https://github.com/discordjs/discord.js/issues/7091) for more context.
+
+### Channel changes
+
+`Channel#createAt` is now nullable. On any regular channel `#createAt` or private thread this value will always be non-null. This value is only nullable for public threads. 
+
+::: tip
+TypeScript users should narrow `Channel` types via type guards in order to get more specific typings.
+:::
+
+### Message Embed Changes
+
+- `MessageEmbed` has now been renamed to `Embed`
+
+- `#setFooter` now accepts a sole object `APIEmbedFooter` object. (add link to dapi site)
```

Comment:
#setFooter takes FooterOptions which accepts camelCase not snake_case

https://github.com/discordjs/discord.js/blob/00ce1c56ac224691a8691a3525cb14ae002319c6/packages/builders/src/messages/embed/Embed.ts#L234